### ISSUE TITLE
can't bind the smex to M-w with ergoemacs-key

### ISSUE THREAD

**Issue by alphaho** [ergoemacs/ergoemacs-mode#105]
Hi Matt,

I have found such issue when I try to swap the functions under M-w and M-a (to reduce using my pinky):

When I loaded `ergoemacs-mode` and `smex`, and tried to bind  `M-W` to `shell-command` and `M-w` to `smex` or `(smex execute-extended-command)`, then I press `M-w`, it will only call `shell-mode` rather than `smex`. 

But if I set `M-w` to `execute-extended-command` only, or set to `(amex execute-extended-command)` but with `smex` not loaded, `M-w` can call `execute-extended-command` without any problem.

I also tried such bindings on `M-t` `M-p` `M-q`, and all these three work. With `M-t` `M-p` `M-q` calling `smex`, and `M-T` `M-P` `M-Q` calling `shell-command`

I can reproduce the issue with a clean startup and a few settings:

`emacs -q  -l ~/.emacs.d/ergo-test.el &`
(Running with: 
  GNU Emacs 24.3.50.1 (x86_64-apple-darwin, NS apple-appkit-1038.36)
  on Mac OS X 10.8.5
  ergoemacs-mode from Github commit cb886a10e13d23c637cbca2c0d1f81648691f026)

**Here's my ergo-test.el:**

``` emacs-lisp

(define-key key-translation-map (kbd "<f13>") (kbd "<menu>"))

(add-to-list 'load-path "/Users/alphaho/ergoemacs-mode/")

(add-to-list 'load-path "/Users/alphaho/.emacs.d/elpa/smex-20130707.1255")

(require 'smex)

(setq debug-on-error t) ; Activate debugging
(require 'ergoemacs-mode)
(setq ergoemacs-debug t)
(setq ergoemacs-keyboard-layout "dv") ; Layout you use.
(cua-mode 0)

;; ;; ergoemacs

(ergoemacs-deftheme alpha
  "my personal theme"
  guru


;; happens with M-w/W
;; smex/(semx execute-extended-command) not work when smex loaded
;; works with M-n/N
;; l,p works
  (ergoemacs-key "M-P" 'shell-command)
  (ergoemacs-key "M-p" '(smex execute-extended-command))  ;; support one of these function

  (ergoemacs-key "M-W" 'shell-command)
  ;(ergoemacs-key "M-w" '(smex execute-extended-command))  ;; toggle this line and the one below doesn't work either
  (ergoemacs-key "M-w" 'smex)


  (ergoemacs-key "M-Q" 'shell-command)
  (ergoemacs-key "M-q" '(smex execute-extended-command))

  (ergoemacs-key "M-T" 'shell-command)
  (ergoemacs-key "M-t" '(smex execute-extended-command))


  (ergoemacs-key "<menu> k" 'end-of-buffer)
  )

(setq ergoemacs-theme "alpha")

(ergoemacs-mode 1)


(provide 'ergo-test)
```

**Comment by laincortes** [comment]
Well, I have some issues related to this when I try to use "M-a" as replace of "M-x" (execute-extended-command). It seems like it hasn't been fixed, or it was introduced again.

**Comment by mattfidler** [comment]
Please post a reproducible example, otherwise I cannot try to fix it.

**Comment by laincortes** [comment]
@mattfidler I was wrong, and I forgot to inform it. It seems that it is triggered when you try to use IDO mode. There is some bad assignment of the variables, or that the mode makes to behave weird the keyboard layout...

tl;dr: I was wrong, it's not ErgoEmacs' fault.

**Comment by mattfidler** [comment]
No problem.  I understand.  It is very difficult to track down.  I use `ido-mode` all the time.  Perhaps if it still gives you a problem, let me know.

**Comment by laincortes** [comment]
It does give me problems. How to reproduce it:

1. M-a->ido-mode (it works until here, it accepts the command with no problem at all).
2. M-a->"Wrong type argument: commandp, smex" (<- that's the output).

I don't know if there is need of more info, if it does, I've no problem to give it.

**Comment by mattfidler** [comment]
Please provide your configuration

**Comment by laincortes** [comment]
OK, this is really embarrassing. I was reading all the packages and *.el; but  didn't read my init.el; until now. It was my fault. I "defadvice-ed" smex in my init.el. So sorry for making you waste your time. :/

**Comment by mattfidler** [comment]
No worries, thanks for the update!

On Mon, Apr 10, 2017, 9:10 PM Lain Cortés González <notifications@github.com>
wrote:

OK, this is really embarrassing. I was reading all the packages and *.el;
but didn't read my init.el; until now. It was my fault. I "defadvice-ed"
smex in my init.el. So sorry for making you waste your time. :/

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ergoemacs/ergoemacs-mode/issues/105#issuecomment-293129433>,
or mute the thread
